Sources/NIO/ByteBuffer-views.swift

```diff
@@ -84,6 +84,26 @@ public struct ByteBufferView: RandomAccessCollection {
             return try body(bytes.bindMemory(to: UInt8.self))
         }
     }
+
+    public func _customIndexOfEquatableElement(_ element : Element) -> Index?? {
+        let indexInRange = self.withUnsafeBytes { (ptr) -> Index? in
```

Comment:
I think this could be written more easily as

    return .some(self.withUnsafeBytes { ptr -> Index? in
        return ptr.firstIndex(of: element).map { $0 + self._range.lowerBound }
    })

but the generated code should be identical :).

Reply:
Agree - your way is much neater.  I will adopt.